### PR TITLE
Bignum: Add BignumModRaw Operation base classes

### PR DESCRIPTION
**Pull request by minosgalanakis**
## Description

This pr is implementing some required base classes ( **BignumModRawOperation**, **BignumModRawOperationArchSplit** )for `bignum_mod_raw.py module`.  

The class structure is configured in order to make is simpler for test classes to inherit from it.

* Adds new optional parameters, which dynamically calculate or convert members if requested.
* Adjusts the expected input variable pair from `(val_a, val_b)` to `(val_n, val_a, val_b(optional)`)
* Added the optional `bits_to_limbs` argument which allows generation of 32 and 64 bit tests cases.
* Added zeropadded string hex representations of all input arguments.

## Gatekeeper checklist

- [x] **changelog** not needed
- [x] **backport** not required
- [x] **tests** not required

## Notes for the submitter

Please refer to the [contributing guidelines](../CONTRIBUTING.md), especially the
checklist for PR contributors.

### CODE REVIEW

File: scripts/generate_driver_wrappers.py
Comment (gilles-peskine-arm) [inline scripts/generate_driver_wrappers.py:29]: Why ignore for typing? We require `types-jsonschema` so mypy should be happy.

File: scripts/mbedtls_dev/bignum_mod_raw.py
Comment (yanesca) [inline scripts/mbedtls_dev/bignum_mod_raw.py]: Why do we need this `pass` here?

File: scripts/mbedtls_dev/bignum_mod_raw.py
Comment (minosgalanakis) [reply scripts/mbedtls_dev/bignum_mod_raw.py:102]: Shall we opt it for the same notation as in the c code? `r_sqrt` -> `rr`?

File: scripts/mbedtls_dev/bignum_mod_raw.py
Comment (minosgalanakis) [reply scripts/mbedtls_dev/bignum_mod_raw.py:54]: I agree. 

The idea was to follow it up with a move out of scaffolding pr. I will update this accordingly.

File: scripts/mbedtls_dev/bignum_mod_raw.py
Comment (yanesca) [reply scripts/mbedtls_dev/bignum_mod_raw.py:102]: In the code we used to use `rr` but in the new code we started moving to `r2`.

File: scripts/mbedtls_dev/bignum_mod_raw.py
Comment (yanesca) [reply scripts/mbedtls_dev/bignum_mod_raw.py]: Never mind...

File: scripts/mbedtls_dev/bignum_mod_raw.py
Comment (minosgalanakis) [reply scripts/mbedtls_dev/bignum_mod_raw.py:102]: r2 it is then !

**Review by yanesca — CHANGES_REQUESTED** [review]
I think it looks good, very nice improvement, like the design a lot.

I can approve once it has been moved out from the merge slot and passes the CI with the `jsonschema` silencing removed.

**Review by yanesca — APPROVED** [review]
LGTM.

**Review by wernerlewis — APPROVED** [review]
LGTM

**Comment by gstrauss** [comment]
FYI: #5396 or #6577 broke my build environment due to a new dependency for the build.

I had to install the python3-jsonschema package on my system.

**Comment by minosgalanakis** [comment]
> FYI: #5396 or #6577 broke my build environment due to a new dependency for the build.
> 
> I had to install the python3-jsonschema package on my system.

That would be the driver dispatch pr which is introducing this new [dependency](https://github.com/Mbed-TLS/mbedtls/pull/5396/files#diff-0a73257cd17cb47b41defecf53fb6c32cca73d451b3d9feb604f00adf6d7db6e). It also updates the [changelog ](https://github.com/Mbed-TLS/mbedtls/pull/5396/files#diff-436ea62ae55b2ab52aef7e465f31dcfe33c070b268457e9c516a13207bcc3b92) accordingly

**Comment by gstrauss** [comment]
@minosgalanakis as discussed in #5396 starting at this [comment](https://github.com/Mbed-TLS/mbedtls/pull/5396#issuecomment-1312554643), the new requirement was not immediately obvious.  The jsonschema requirement is listed in [docs/proposed/psa-driver-wrappers-codegen-migration-guide.md](https://github.com/Mbed-TLS/mbedtls/pull/5396/files#diff-436ea62ae55b2ab52aef7e465f31dcfe33c070b268457e9c516a13207bcc3b92) and [scripts/driver.requirements.txt](https://github.com/Mbed-TLS/mbedtls/pull/5396/files#diff-0a73257cd17cb47b41defecf53fb6c32cca73d451b3d9feb604f00adf6d7db6e)  The changelog entry does not list the new requirement.  Please consider listing new requirements in the changelog so that it is more easily findable by those who did not specifically review the commit in detail.  Thank you.

**Pull request merged**
